File: hw/basurto/u1/hw07Collections/Python/view/collections_app.py
from datetime import date
from model.person import Person


def print_the_collection(things):
    print(f"\nsize of things --> {len(things)}")
    print(f"These are my things --> \n{things}")


def main():

    id = 1
    full_name = "Daniel Codena"
    born_on_date = date(2006, 5, 15)
    alive = True

    person = Person(id, full_name, born_on_date, alive)

    print(person)
    print("Age of person 1 is", person.compute_age_in_years())

    id = 2
    full_name = "Eustaquio Perez de la Fuente Pereira"
    born_on_date = date(1990, 12, 11)
    alive = False

    person2 = Person(id, full_name, born_on_date, alive)

    print("Age of person 2 is",
          person2.compute_age_in_years())

    things = []

    print("size of things -->", things)

    things.append(8000)
    things.append(3.5)
    things.append("Quito")
    things.append(True)

    print_the_collection(things)

    things.append(person)
    things.append(person2)

    print_the_collection(things)

    things.append(4078.76)
    things.append("a")

    print_the_collection(things)

    people = []

    people.append(person)
    people.append(person2)

    print("\npeople -->", people)
    print("")

    for p in people:
        print("Person -->", p)


if __name__ == "__main__":
    main()

File: hw/basurto/u1/hw07Collections/JavaScript/view/collections_app.js
const Person = require('../model/person');

function printTheCollection(things) {

    console.log(
        "\nsize of things --> " +
        things.length
    );

    console.log(
        "These are my things -->"
    );

    things.forEach(item => {

        if (item instanceof Person) {
            console.log(item.toString());
        } else {
            console.log(item);
        }

    });
}

function main() {

    let id;
    let fullName;
    let bornOnDate;
    let alive;

    id = 1;
    fullName = "Daniel Codena";
    bornOnDate = new Date(2006, 4, 15);
    alive = true;

    const person =
        new Person(
            id,
            fullName,
            bornOnDate,
            alive
        );

    console.log(person.toString());

    console.log(
        "Age of person 1 is",
        person.computeAgeInYears()
    );

    id = 2;
    fullName =
        "Eustaquio Perez de la Fuente Pereira";

    bornOnDate = new Date(1990, 11, 11);

    alive = false;

    const person2 =
        new Person(
            id,
            fullName,
            bornOnDate,
            alive
        );

    console.log(
        "Age of person 2 is",
        person2.computeAgeInYears()
    );

    let things = [];

    console.log(
        "size of things -->",
        things
    );

    things.push(8000);
    things.push(3.5);
    things.push("Quito");
    things.push(true);

    printTheCollection(things);

    things.push(person);
    things.push(person2);

    printTheCollection(things);

    things.push(4078.76);
    things.push("a");

    printTheCollection(things);

    let people = [];

    people.push(person);
    people.push(person2);

    console.log("\npeople -->", people);

    people.forEach((p) => {
        console.log("Person -->",
            p.toString());
    });
}

main();